test/unit/helpers/httpApi.js

```diff
@@ -0,0 +1,488 @@
+					};
+					var config = {
+						peers: {
+							enabled: false,
+							access: {
+								blackList: false
+							}
+						}
+					};
+					var next = function () {return true;};
+					var res = {
+						status: function (httpCode) {
+							return {
+								send: function (obj) {
+
+								}
+							};
+						}
+					};
+					httpApi.middleware.applyAPIAccessRules(config, req, res, next);
+				});
+
+				it('does not matches the peer url does not call next', function () {
+					var sendCalled = false;
+					var nextCalled = false;
+					var matchCalled = false;
+					var req = {
+						url: {
+							match: function (regex) {
+								matchCalled = true;
+								return false;
+							}
+						},
+						ip: '127.0.0.1'
+					};
+					var config = {
+						api: {
+							enabled: true,
+							access: {
+								public: false
+							}
+						}
+					};
+					var next = function () {
+						nextCalled = true;
+						return true;
+					};
+					var res = {
+						status: function (httpCode) {
+							return {
+								send: function (obj) {
+									expect(obj).to.eql({success:false,error:'API access denied'});
+									sendCalled = true;
+								}
+							};
+						}
+					};
+					httpApi.middleware.applyAPIAccessRules(config, req, res, next);
+					expect(sendCalled).to.be.true;
+					expect(nextCalled).to.be.false;
+					expect(matchCalled).to.be.true;
+				});
+
+				it('does not matches the peer url does call next', function () {
+					var sendCalled = false;
+					var nextCalled = false;
+					var matchCalled = false;
+					var req = {
+						url: {
+							match: function (regex) {
+								matchCalled = true;
+								return false;
+							}
+						},
+						ip: '127.0.0.1'
+					};
+					var config = {
+						api: {
+							enabled: true,
+							access: {
+								public: true
+							}
+						}
+					};
+					var next = function () {
+						nextCalled = true;
+						return true;
+					};
+					var res = {
+						status: function (httpCode) {
+							return {
+								send: function (obj) {
+									sendCalled = true;
+								}
+							};
+						}
+					};
+					httpApi.middleware.applyAPIAccessRules(config, req, res, next);
+					expect(sendCalled).to.be.false;
+					expect(nextCalled).to.be.true;
+					expect(matchCalled).to.be.true;
+				});
+			});
+		});
+	});
+
+	describe('registerEndpoint', function () {
+		it('registers the endpoint', function () {
+			var called = 0;
+			var appUseCalled = false;
+			var router = {
+				use: function (afunction) {
+					expect(_.isFunction(afunction)).to.be.true;
+					called += 1;
+				},
+				mine: true
+			};
+			var app = {
+				use: function (aroute, arouter) {
+					expect(aroute).to.equal('route');
+					expect(arouter.mine).to.be.true;
+					appUseCalled = true;
+				}
+			};
+			httpApi.registerEndpoint('route', app, router, false);
+			expect(called).to.equal(2);
+			expect(appUseCalled).to.be.true;
+		});
+	});
+
+	describe('respond', function () {
```

Comment:
Please basing on the comments left to make the test more descriptive and specific